qubes/vm/mix/dvmtemplate.py

```diff
@@ -265,6 +264,13 @@ async def on_domain_preload_dispvm_used(
             event_log += " because %s" % str(kwargs.get("reason"))
         self.log.info(event_log)
 
+        service = "qubes.WaitForRunningSystem"
+        if not self.supports_preload():
+            raise qubes.exc.QubesValueError(
+                "Qube does not support the RPC '%s' but tried to preload, "
+                "check if template is outdated" % service
+            )
```

Comment:
This should be after possibly removing old preloaded below, no? Otherwise, you may end up with outdated preloaded disposables, especially when connected with https://github.com/QubesOS/qubes-core-admin/pull/707 (if it just lost that service support for some reason).